### ISSUE TITLE
Error: org.jpmml.evaluator.InvalidResultException

### ISSUE THREAD

**Issue by s2chen** [jpmml/jpmml-evaluator#97]
Hello Villu,

When I evaluated the pipleline, I am getting this InvalidResultException error. From your other posts, you indicated this might be due to the data is outside of the ranges indicated in the PMML file. However, since I am using the same data for training and testing, this shouldn't occur. Could you please take a look at what is wrong? Any help is appreciated

```python
import random
import pandas as pd
from sklearn.preprocessing import StandardScaler,LabelEncoder
from sklearn2pmml import PMMLPipeline,sklearn2pmml
from sklearn2pmml.decoration import CategoricalDomain, ContinuousDomain
from sklearn_pandas import DataFrameMapper
from sklearn.cluster import KMeans

path = '/Users/shchen/Desktop/'
    
data = pd.read_csv(path+'data.csv')

features_normalize = ['Val1','Val3']

mapper = DataFrameMapper(
        [([column], [ContinuousDomain(),StandardScaler()]) for column in features_normalize] + 
        [(['Val2'],[ContinuousDomain()])] +
        [(['Category'],[CategoricalDomain()])] +
        [(['Label'],[CategoricalDomain(),LabelEncoder()])])

kmeans = KMeans(init='k-means++', n_clusters=2, n_init=20)    
pipeline = PMMLPipeline([
    ("mapper",mapper),
    ("clusterer", kmeans)
])   
pipeline.fit(data,y=None)

sklearn2pmml(pipeline, path+'test.pmml', with_repr = True)
```

This is some made up data

Label | Category | Val1 | Val2 | Val3
-- | -- | -- | -- | --
Email | 1 | 73.2 | 0.75 | 78
Password | 0 | 43.73 | 0.73 | 83
Ingestion | 0 | 4.51 | 0.57 | 65
ATT | 0 | 89.64 | 0.83 | 41
Password | 1 | 73.05 | 0.26 | 85
Ingestion | 1 | 31.48 | 0.02 | 89
ATT | 0 | 0.72 | 0.24 | 21
Email | 1 | 71.98 | 0.29 | 31
Password | 0 | 35.81 | 0.16 | 73
Ingestion | 0 | 23.12 | 0.32 | 21

This is the error
```java
Global Error: com.ea.eadp.risk.pmml.PMMLException: Failed to create PMML evaluator
	at com.ea.eadp.risk.pmml.impl.PMMLServiceImpl.createEvaluator(PMMLServiceImpl.java:92)
	at com.ea.eadp.risk.pmml.impl.loadtest.PMMLLoadTest.createEvaluator(PMMLLoadTest.java:221)
	at com.ea.eadp.risk.pmml.impl.loadtest.PMMLLoadTest.evaluate(PMMLLoadTest.java:82)
	at com.ea.eadp.risk.pmml.impl.loadtest.PMMLLoadTestServiceImpl.runLoadTest(PMMLLoadTestServiceImpl.java:20)
	at com.ea.eadp.test.jpmml.Program.runLoadTest(Program.java:138)
	at com.ea.eadp.test.jpmml.Program.main(Program.java:54)
Caused by: org.jpmml.evaluator.InvalidResultException
	at org.jpmml.evaluator.FieldValueUtil.performInvalidValueTreatment(FieldValueUtil.java:190)
	at org.jpmml.evaluator.FieldValueUtil.prepareInputValue(FieldValueUtil.java:94)
	at org.jpmml.evaluator.InputField.prepare(InputField.java:64)
	at org.jpmml.evaluator.EvaluatorUtil.prepare(EvaluatorUtil.java:123)
	at com.ea.eadp.risk.pmml.impl.PMMLEvaluatorImpl.evaluate(PMMLEvaluatorImpl.java:152)
	at com.ea.eadp.risk.pmml.impl.PMMLServiceImpl.createEvaluator(PMMLServiceImpl.java:118)
	at com.ea.eadp.risk.pmml.impl.PMMLServiceImpl.createEvaluator(PMMLServiceImpl.java:69)
```
I have also tried to remove this under DataField for Val1, Val2, Val3, but they didn't help

```xml
<Interval closure="closedClosed" leftMargin="0.72" rightMargin="89.64"/>
<Interval closure="closedClosed" leftMargin="21.0" rightMargin="89.0"/>
<Interval closure="closedClosed" leftMargin="0.02" rightMargin="0.83"/>
```

This is the entirety of the pmml output file

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<PMML xmlns="http://www.dmg.org/PMML-4_3" version="4.3">
	<Header>
		<Application name="JPMML-SkLearn" version="1.4.0"/>
		<Timestamp>2018-01-25T23:23:10Z</Timestamp>
	</Header>
	<MiningBuildTask>
		<Extension>PMMLPipeline(steps=[('mapper', DataFrameMapper(default=False, df_out=False,
        features=[(['Val1'], [ContinuousDomain(), StandardScaler(copy=True, with_mean=True, with_std=True)]), (['Val3'], [ContinuousDomain(), StandardScaler(copy=True, with_mean=True, with_std=True)]), (['Val2'], [ContinuousDomain()]), (['Category'], [CategoricalDomain()]), (['Label'], [CategoricalDomain(), LabelEncoder()])],
        input_df=False, sparse=False)),
       ('clusterer', KMeans(algorithm='auto', copy_x=True, init='k-means++', max_iter=300,
    n_clusters=2, n_init=20, n_jobs=1, precompute_distances='auto',
    random_state=None, tol=0.0001, verbose=0))])</Extension>
	</MiningBuildTask>
	<DataDictionary>
		<DataField name="Val1" optype="continuous" dataType="double">
			<Interval closure="closedClosed" leftMargin="0.72" rightMargin="89.64"/>
		</DataField>
		<DataField name="Val3" optype="continuous" dataType="double">
			<Interval closure="closedClosed" leftMargin="21.0" rightMargin="89.0"/>
		</DataField>
		<DataField name="Val2" optype="continuous" dataType="double">
			<Interval closure="closedClosed" leftMargin="0.02" rightMargin="0.83"/>
		</DataField>
		<DataField name="Category" optype="continuous" dataType="integer">
			<Value value="0"/>
			<Value value="1"/>
		</DataField>
		<DataField name="Label" optype="categorical" dataType="string">
			<Value value="ATT"/>
			<Value value="Email"/>
			<Value value="Ingestion"/>
			<Value value="Password"/>
		</DataField>
	</DataDictionary>
	<TransformationDictionary>
		<DerivedField name="standard_scaler(Val1)" optype="continuous" dataType="double">
			<Apply function="/">
				<Apply function="-">
					<FieldRef field="Val1"/>
					<Constant dataType="double">44.724000000000004</Constant>
				</Apply>
				<Constant dataType="double">29.410343486603484</Constant>
			</Apply>
		</DerivedField>
		<DerivedField name="standard_scaler(Val3)" optype="continuous" dataType="double">
			<Apply function="/">
				<Apply function="-">
					<FieldRef field="Val3"/>
					<Constant dataType="double">58.7</Constant>
				</Apply>
				<Constant dataType="double">25.961702563583923</Constant>
			</Apply>
		</DerivedField>
		<DerivedField name="label_encoder(Label)" optype="continuous" dataType="integer">
			<MapValues outputColumn="output">
				<FieldColumnPair field="Label" column="input"/>
				<InlineTable>
					<row>
						<input xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">ATT</input>
						<output xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">0</output>
					</row>
					<row>
						<input xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">Email</input>
						<output xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">1</output>
					</row>
					<row>
						<input xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">Ingestion</input>
						<output xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">2</output>
					</row>
					<row>
						<input xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">Password</input>
						<output xmlns:ns2="http://www.dmg.org/PMML-4_3" xmlns="">3</output>
					</row>
				</InlineTable>
			</MapValues>
		</DerivedField>
	</TransformationDictionary>
	<ClusteringModel functionName="clustering" modelClass="centerBased" numberOfClusters="2">
		<MiningSchema>
			<MiningField name="Val2" missingValueTreatment="asIs"/>
			<MiningField name="Category" missingValueTreatment="asIs"/>
			<MiningField name="Val1" missingValueTreatment="asIs"/>
			<MiningField name="Val3" missingValueTreatment="asIs"/>
			<MiningField name="Label" missingValueTreatment="asIs"/>
		</MiningSchema>
		<Output>
			<OutputField name="Cluster" optype="categorical" dataType="string" feature="predictedValue"/>
			<OutputField name="affinity(0)" optype="continuous" dataType="double" feature="affinity" value="0"/>
			<OutputField name="affinity(1)" optype="continuous" dataType="double" feature="affinity" value="1"/>
		</Output>
		<ModelStats>
			<UnivariateStats field="Val1">
				<Counts totalFreq="10.0" missingFreq="0.0" invalidFreq="0.0"/>
				<NumericInfo minimum="0.72" maximum="89.64" mean="44.724000000000004" standardDeviation="29.410343486603484" median="39.769999999999996" interQuartileRange="47.5725"/>
			</UnivariateStats>
			<UnivariateStats field="Val3">
				<Counts totalFreq="10.0" missingFreq="0.0" invalidFreq="0.0"/>
				<NumericInfo minimum="21.0" maximum="89.0" mean="58.7" standardDeviation="25.961702563583923" median="69.0" interQuartileRange="48.25"/>
			</UnivariateStats>
			<UnivariateStats field="Val2">
				<Counts totalFreq="10.0" missingFreq="0.0" invalidFreq="0.0"/>
				<NumericInfo minimum="0.02" maximum="0.83" mean="0.417" standardDeviation="0.2660845730214362" median="0.305" interQuartileRange="0.44499999999999995"/>
			</UnivariateStats>
			<UnivariateStats field="Category">
				<Counts totalFreq="10.0" missingFreq="0.0" invalidFreq="0.0"/>
				<DiscrStats>
					<Array type="string">0 1</Array>
					<Array type="int">6 4</Array>
				</DiscrStats>
			</UnivariateStats>
			<UnivariateStats field="Label">
				<Counts totalFreq="10.0" missingFreq="0.0" invalidFreq="0.0"/>
				<DiscrStats>
					<Array type="string">ATT Email Ingestion Password</Array>
					<Array type="int">2 2 3 3</Array>
				</DiscrStats>
			</UnivariateStats>
		</ModelStats>
		<ComparisonMeasure kind="distance" compareFunction="absDiff">
			<squaredEuclidean/>
		</ComparisonMeasure>
		<ClusteringField field="standard_scaler(Val1)"/>
		<ClusteringField field="standard_scaler(Val3)"/>
		<ClusteringField field="Val2"/>
		<ClusteringField field="Category"/>
		<ClusteringField field="label_encoder(Label)"/>
		<Cluster id="0" size="6">
			<Array type="real">-0.03719779745171368 0.775501270921039 0.415 0.5 2.333333333333333</Array>
		</Cluster>
		<Cluster id="1" size="4">
			<Array type="real">0.0557966961775703 -1.163251906381559 0.42 0.25 0.75</Array>
		</Cluster>
	</ClusteringModel>
</PMML>
```

**Comment by vruusmann** [comment]
Your model scores your sample data just fine:

(null) | Cluster | affinity(0) | affinity(1)
-- | -- | -- | --
0 | 0 | 3.1519197325616677 | 5.201767788858241
0 | 0 | 0.8194389484870619 | 9.635960870350377
0 | 0 | 2.4383342522160927 | 5.649426674401318
1 | 1 | 10.437715263418486 | 3.1900015434118987
0 | 0 | 1.775546425075952 | 11.210060398186315
0 | 0 | 0.8411567035752625 | 7.971709252889073
1 | 1 | 12.816162127480204 | 3.149575240018495
1 | 1 | 6.367245386833517 | 1.4097303949902873
0 | 0 | 0.8806540193708581 | 8.259412913345951
1 | 1 | 5.818847225446061 | 2.343137613594671
